src/Kiota.Builder/CodeRenderer.cs

```diff
@@ -28,28 +19,38 @@ public static async Task RenderCodeNamespaceToSingleFileAsync(LanguageWriter wri
             RenderCode(writer, codeElement);
             await sw.FlushAsync();
         }
-
-        public static async Task RenderCodeNamespaceToFilePerClassAsync(LanguageWriter writer, CodeNamespace root)
+        // We created barrells for codenamespaces. Skipping for empty namespaces, ones created for users, and ones with same namspace as class name.
+        public static async Task RenderCodeNamespaceToFilePerClassAsync(LanguageWriter writer, CodeNamespace root, bool shouldWriteNamespaceIndices, string namespacePrefix)
         {
             foreach (var codeElement in root.GetChildElements(true))
             {
                 if (codeElement is CodeClass codeClass)
                     await RenderCodeNamespaceToSingleFileAsync(writer, codeClass, writer.PathSegmenter.GetPath(root, codeClass));
                 else if (codeElement is CodeEnum codeEnum)
                     await RenderCodeNamespaceToSingleFileAsync(writer, codeEnum, writer.PathSegmenter.GetPath(root, codeEnum));
-                else if(codeElement is CodeNamespace codeNamespace)
-                    await RenderCodeNamespaceToFilePerClassAsync(writer, codeNamespace);
+                else if(codeElement is CodeNamespace codeNamespace) {
+                    
+                    if(!string.IsNullOrEmpty(codeNamespace.Name) && !string.IsNullOrEmpty(root.Name) && shouldWriteNamespaceIndices && !namespacePrefix.Contains(codeNamespace.Name, StringComparison.OrdinalIgnoreCase)) {
```

Comment:
I didn't see class didn't import System, this is why it's failing.
Please add  `using System;` at the top of this file.